src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -139,8 +146,10 @@ - (void)finishWithError:(NSError *)errorOrNil {
   _self = nil;
 
   // If there were still request messages coming, stop them.
-  _requestWriter.state = GRXWriterStateFinished;
-  _requestWriter = nil;
+  @synchronized(_requestWriter) {
```

Comment:
This why @property defaults to atomic - 
@property GRXWriter *requestWriter;

...

@synchronized(self.requestWriter) {...}

I'm under the impression that the compiler essentially generates:
- (GRXWriter *)requestWriter {
  @synchronized(self) {
    return [[_requestWriter retain] autorelease];
  }
}

- (void)setRequestWriter:(GRXWriter *)requestWriter {
  @synchronized(self) {
    if (_requestWriter != requestWriter) {
      [_requestWriter autorelease];
      _requestWriter = [requestWriter retain];
    }
  }
}

so that other threads can't look at a partially deallocated object if the setter would otherwise cause the old value to be deallocated.